Core/Core/API/API.swift

```diff
@@ -112,10 +128,43 @@ public struct URLSessionAPI: API {
         #endif
         return urlSession.uploadTask(with: request, fromFile: url)
     }
+
+    func refreshToken(callback: @escaping () -> Void) {
+        guard
+            let loginSession = loginSession,
+            let refreshToken = loginSession.refreshToken,
+            let clientID = loginSession.clientID,
+            let clientSecret = loginSession.clientSecret
+        else {
+            return callback()
+        }
+        let client = APIVerifyClient(authorized: true, base_url: baseURL, client_id: clientID, client_secret: clientSecret)
+        let request = PostLoginOAuthRequest(client: client, refreshToken: refreshToken)
+        makeRequest(request, refreshToken: false) { [weak self] response, _, error in
+            if let response = response, error == nil {
+                let session = loginSession.refresh(accessToken: response.access_token)
+                LoginSession.add(session)
```

Comment:
Should it also update the AppEnvironment.currentSession if it matches?

Reply:
Yes, good call!